File: smartmoneyserver/src/main/java/fpt/aptech/server/enums/category/SystemCategory.java
package fpt.aptech.server.enums.category;

import lombok.Getter;

@Getter
public enum SystemCategory {
    
    // ================= CHI TIÊU (EXPENSE) =================
    FOOD_AND_BEVERAGE(1),       // Ăn uống
    INSURANCE(2),               // Bảo hiểm
    OTHER_EXPENSE(3),           // Các chi phí khác
    INVESTMENT(4),              // Đầu tư
    TRANSPORTATION(5),          // Di chuyển
    FAMILY(6),                  // Gia đình
    ENTERTAINMENT(7),           // Giải trí
    EDUCATION(8),               // Giáo dục
    BILLS_AND_UTILITIES(9),     // Hoá đơn & Tiện ích
    SHOPPING(10),               // Mua sắm
    GIFTS_AND_DONATIONS(11),    // Quà tặng & Quyên góp
    HEALTH(12),                 // Sức khỏe
    TRANSFER_OUT(13),           // Tiền chuyển đi
    INTEREST_PAYMENT(14),       // Trả lãi
    
    // ================= THU NHẬP (INCOME) =================
    SALARY(15),                 // Lương
    INTEREST_RECEIVE(16),       // Thu lãi
    INCOME_OTHER(17),           // Thu nhập khác (Dùng khi tạo Wallet)
    INCOME_TRANSFER(18),        // Tiền chuyển đến (Dùng khi tạo Saving Goal)
    
    // ================= NỢ & VAY (DEBT & LOAN) =================
    DEBT_LENDING(19),           // Cho vay
    DEBT_BORROWING(20),         // Đi vay
    DEBT_COLLECTION(21),        // Thu nợ
    DEBT_REPAYMENT(22),         // Trả nợ

    // ================= DANH MỤC CON (SUB-CATEGORIES) =================
    CAR_MAINTENANCE(23),        // Bảo dưỡng xe
    HOME_SERVICES(24),          // Dịch vụ gia đình
    HOME_DECOR(25),             // Sửa & trang trí nhà
    PETS(26),                   // Vật nuôi
    ONLINE_SERVICES(27),        // Dịch vụ trực tuyến
    TRAVEL(28),                 // Vui - chơi
    ELECTRICITY_BILL(29),       // Hoá đơn điện
    PHONE_BILL(30),             // Hoá đơn điện thoại
    GAS_BILL(31),               // Hoá đơn gas
    INTERNET_BILL(32),          // Hoá đơn internet
    WATER_BILL(33),             // Hoá đơn nước
    OTHER_UTILITY_BILL(34),     // Hoá đơn tiện ích khác
    TV_BILL(35),                // Hoá đơn TV
    RENTAL(36),                 // Thuê nhà
    PERSONAL_ITEMS(37),         // Đồ dùng cá nhân
    HOME_APPLIANCES(38),        // Đồ gia dụng
    BEAUTY(39),                 // Làm đẹp
    MEDICAL_CHECKUP(40),        // Khám sức khoẻ
    SPORTS(41);                 // Thể dục thể thao

    private final int id;

    SystemCategory(int id) {
        this.id = id;
    }
}

File: APP MOBILE/smart_money/lib/modules/category/enums/system_category.dart
/// 36 danh mục hệ thống: Ăn uống, Bảo hiểm, Giao thông, v.v.
enum SystemCategory {
  // Chi tiêu
  foodAndBeverage,      // 1
  insurance,            // 2
  otherExpense,         // 3
  investment,           // 4
  transportation,       // 5
  family,               // 6
  entertainment,        // 7
  education,            // 8
  billsAndUtilities,    // 9
  shopping,             // 10
  giftsAndDonations,    // 11
  health,               // 12
  transferOut,          // 13
  interestPayment,      // 14

  // Thu nhập
  salary,               // 15
  interestReceive,      // 16
  incomeOther,          // 17
  incomeTransfer,       // 18

  // Nợ & Vay
  debtLending,          // 19
  debtBorrowing,        // 20
  debtCollection,       // 21
  debtRepayment,        // 22

  // Danh mục con
  carMaintenance,       // 23
  homeServices,         // 24
  homeDecor,            // 25
  pets,                 // 26
  onlineServices,       // 27
  travel,               // 28
  electricityBill,      // 29
  phoneBill,            // 30
  gasBill,              // 31
  internetBill,         // 32
  waterBill,            // 33
  otherUtilityBill,     // 34
  tvBill,               // 35
  rental;               // 36

  int get value => index + 1;

  static SystemCategory fromValue(int value) {
    if (value < 1 || value > 36) {
      return SystemCategory.otherExpense;
    }
    return SystemCategory.values[value - 1];
  }

  String get displayName {
    switch (this) {
      case SystemCategory.foodAndBeverage:
        return 'Food & Beverage';
      case SystemCategory.insurance:
        return 'Insurance';
      case SystemCategory.otherExpense:
        return 'Other Expense';
      case SystemCategory.investment:
        return 'Investment';
      case SystemCategory.transportation:
        return 'Transportation';
      case SystemCategory.family:
        return 'Family';
      case SystemCategory.entertainment:
        return 'Entertainment';
      case SystemCategory.education:
        return 'Education';
      case SystemCategory.billsAndUtilities:
        return 'Bills & Utilities';
      case SystemCategory.shopping:
        return 'Shopping';
      case SystemCategory.giftsAndDonations:
        return 'Gifts & Donations';
      case SystemCategory.health:
        return 'Health';
      case SystemCategory.transferOut:
        return 'Transfer Out';
      case SystemCategory.interestPayment:
        return 'Interest Payment';
      case SystemCategory.salary:
        return 'Salary';
      case SystemCategory.interestReceive:
        return 'Interest Received';
      case SystemCategory.incomeOther:
        return 'Other Income';
      case SystemCategory.incomeTransfer:
        return 'Transfer In';
      case SystemCategory.debtLending:
        return 'Lending';
      case SystemCategory.debtBorrowing:
        return 'Borrowing';
      case SystemCategory.debtCollection:
        return 'Debt Collection';
      case SystemCategory.debtRepayment:
        return 'Debt Repayment';
      case SystemCategory.carMaintenance:
        return 'Car Maintenance';
      case SystemCategory.homeServices:
        return 'Home Services';
      case SystemCategory.homeDecor:
        return 'Home Decor';
      case SystemCategory.pets:
        return 'Pets';
      case SystemCategory.onlineServices:
        return 'Online Services';
      case SystemCategory.travel:
        return 'Travel';
      case SystemCategory.electricityBill:
        return 'Electricity Bill';
      case SystemCategory.phoneBill:
        return 'Phone Bill';
      case SystemCategory.gasBill:
        return 'Gas Bill';
      case SystemCategory.internetBill:
        return 'Internet Bill';
      case SystemCategory.waterBill:
        return 'Water Bill';
      case SystemCategory.otherUtilityBill:
        return 'Other Utility Bill';
      case SystemCategory.tvBill:
        return 'TV Bill';
      case SystemCategory.rental:
        return 'Rental';
    }
  }
}

